components/script/dom/document.rs

```diff
@@ -2232,12 +2233,24 @@ impl Document {
 
     pub fn register_responsive_image(&self, img: &HTMLImageElement) {
         self.responsive_images.borrow_mut().push(Dom::from_ref(img));
+        let elem = img.upcast::<Element>();
+        let is_parent_picture = elem.upcast::<Node>().GetParentElement()
+            .map_or(false, |p| p.is::<HTMLPictureElement>());
+        if is_parent_picture {
+            img.update_the_image_data();
+        }
```

Comment:
I don't think it makes sense to implement these mutation checks here. It splits the implementation between multiple files so it's harder to understand how the pieces fit together.